File: backend/src/main/java/ch/rodano/core/services/bll/configurator/FeatureService.java
package ch.rodano.core.services.bll.configurator;

import java.util.List;
import java.util.UUID;

import ch.rodano.api.config.FeatureDTO;

public interface FeatureService {

	List<FeatureDTO> getFeatures(UUID projectId);

	FeatureDTO getFeature(UUID projectId, UUID featureId);

	FeatureDTO createFeature(UUID projectId, FeatureDTO feature);

	FeatureDTO updateFeature(UUID projectId, UUID featureId, FeatureDTO feature);

	void deleteFeature(UUID projectId, UUID featureId);
}

File: frontends/main/src/app/configurator/services/api/feature.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Feature} from '@core/model/feature';

@Injectable({
	providedIn: 'root'
})
export class FeatureService {
	constructor(private http: HttpClient) {}

	getFeatures(projectId: string): Observable<Feature[]> {
		return this.http.get<Feature[]>(`/api/superuser/configurator/projects/${projectId}/config/features`);
	}

	getFeature(projectId: string, featureId: string): Observable<Feature> {
		return this.http.get<Feature>(`/api/superuser/configurator/projects/${projectId}/config/features/${featureId}`);
	}

	createFeature(projectId: string, feature: Feature): Observable<Feature> {
		return this.http.post<Feature>(`/api/superuser/configurator/projects/${projectId}/config/features`, feature);
	}

	updateFeature(projectId: string, featureId: string, feature: Feature): Observable<Feature> {
		return this.http.put<Feature>(`/api/superuser/configurator/projects/${projectId}/config/features/${featureId}`, feature);
	}

	deleteFeature(projectId: string, featureId: string): Observable<void> {
		return this.http.delete<void>(`/api/superuser/configurator/projects/${projectId}/config/features/${featureId}`);
	}
}
